hack/storage-version-migration.sh

```diff
@@ -65,19 +65,23 @@ spec:
   ttlSecondsAfterFinished: 600
 EOF
 
-JOB_NAME=$(kubectl -n shipwright-build get job --selector app=storage-version-migration-shipwright -o jsonpath='{.items[0].metadata.name}')
 
-while [ "$(kubectl -n shipwright-build get job "${JOB_NAME}" -o json | jq -r '.status.completionTime // ""')" == "" ]; do
-    echo "[INFO] Storage version migraton job is still running"
-    sleep 10
-done
+NAMESPACE="shipwright-build"
+JOB_NAME=$(kubectl -n "${NAMESPACE}" get job --selector app=storage-version-migration-shipwright -o jsonpath='{.items[0].metadata.name}')
 
-isFailed="$(kubectl -n shipwright-build get job "${JOB_NAME}" -o json | jq -r '.status.conditions[] | select(.type == "Failed") | .status')"
+while true; do
+    jobStatus=$(kubectl get job "${JOB_NAME}" -n "${NAMESPACE}" -o=jsonpath='{.status.conditions[*].type}')
 
-if [ "${isFailed}" == "True" ]; then
-    echo "[ERROR] Storage version migration failed"
-    kubectl -n shipwright-build logs "job/${JOB_NAME}"
-    exit 1
-fi
+    if [[ "${jobStatus}" == *"Complete"* ]]; then
+		echo "Job ${JOB_NAME} has completed successfully!"
+		exit 0
```

Comment:
This is not sufficient. `condition.type` tells you if the status condition is reporting the `Complete` or `Failed` . The `Complete` and `Failed` condition types should always be populated in the `Conditions` array.

You need to inspect the `condition.status` field, which can be `True`, `False`, or `Unknown`.